### PR TITLE
add support for selfhosted gitlab/github

### PR DESCRIPTION
**Pull request by Kloenk**
This adds support for a self-hosted GitLab and GitHub (I if the api is the same as on github.com).

### Example
Test repo on [git.thevillage.chat](https://git.thevillage.chat/kloenk/dwarffs)
~~~
./result/bin/nix flake info gitlab:kloenk/dwarffs\?url=git.thevillage.chat
Resolved URL:  gitlab:kloenk/dwarffs
Locked URL:    gitlab:kloenk/dwarffs/799b8e0abc624e440bc35c38f4b5a08ce981df4a
Description:   A filesystem that fetches DWARF debug info from the Internet on demand
Path:          /nix/store/v6xvpisgmnp2qwjbcl5xl735g7md2c6l-source
Revision:      799b8e0abc624e440bc35c38f4b5a08ce981df4a
Last modified: 2020-05-11 19:41:56
~~~

I don't know how to change the output of `nix flake info` to include the host instance.

### CODE REVIEW

**Comment by Kloenk** [comment]
While writing this I noticed that `nix dev-shell` is broken. While following the instructions in the readme, the nix build with this tries to access `/usr/local/` which is not available (NixOS)

File: src/libutil/url.hh [hunk 1]
```diff
@@ -49,6 +49,10 @@ const static std::string pathRegex = "(?:" + segmentRegex + "(?:/" + segmentRege
 const static std::string refRegexS = "[a-zA-Z0-9][a-zA-Z0-9_.-]*"; // FIXME: check
 extern std::regex refRegex;
 
+// A github or gitlab url
+const static std::string urlRegexS = "[a-zA-Z0-9.]*"; // FIXME: check
```
Comment (edolstra) [inline src/libutil/url.hh:53]: The name `urlRegex` does not correspond with that this actually is. Maybe it's better to move this to `github.cc`.

Comment (Kloenk) [reply src/libutil/url.hh]: moved it to `github.cc`

**Comment by Kloenk** [comment]
@edolstra 
I got the instance URL into `nix flake info`. Is this output desired? should I create a new pr for that?

~~~
./result/bin/nix flake info gitlab:kloenk/dwarffs\?url=git.thevillage.chat
Resolved URL:  gitlab:kloenk/dwarffs
Instance URL:  git.thevillage.chat
Locked URL:    gitlab:kloenk/dwarffs/799b8e0abc624e440bc35c38f4b5a08ce981df4a
Description:   A filesystem that fetches DWARF debug info from the Internet on demand
Path:          /nix/store/v6xvpisgmnp2qwjbcl5xl735g7md2c6l-source
Revision:      799b8e0abc624e440bc35c38f4b5a08ce981df4a
Last modified: 2020-05-11 19:41:56
~~~

**Pull request merged**